js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+import Runtime from '../../wasm/publish/rive.js';
+
+// #region LoopEvent
+
+// Loop types. The index of the type is the value that comes from Wasm
+export const loopTypes: string[] = ['oneShot', 'loop', 'pingPong'];
+
+/// Loop events are returned through onloop callbacks
+interface LoopEvent {
+  animation: string;
+  name: string;
+  type: number;
+}
+
+// Creates a new LoopEvent
+export let createLoopEvent = (animation: string, loopValue: number): LoopEvent => {
+  if (loopValue < 0 || loopValue >= loopTypes.length) {
+    throw 'Invalid loop value';
+  }
+  return {
+    animation: animation,
+    type: loopValue,
+    name: loopTypes[loopValue],
+  }
+}
+
+// #endregion
+
+// Maps the playback state to the Wasm enum values
+export const playbackStates = { 'play': 0, 'pause': 1, 'stop': 2 };
+
+// #region layout
+
+// Fit options for the canvas
+export enum Fit {
+  Cover     = 'cover',
+  Contain   = 'contain',
+  Fill      = 'fill',
+  FitWidth  = 'fitWidth',
+  FitHeight = 'fitHeight',
+  None      = 'none',
+  ScaleDown = 'scaleDown'
+}
+
+// Alignment options for the canvas
+export enum Alignment {
+  Center       = 'center',
+  TopLeft      = 'topLeft',
+  TopCenter    = 'topCenter',
+  TopRight     = 'topRight',
+  CenterLeft   = 'centerLeft',
+  CenterRight  = 'centerRight',
+  BottomLeft   = 'bottomLeft',
+  BottomCenter = 'bottomCenter',
+  BottomRight  = 'bottomRight'
+}
+
+// Interface for the Layout static method contructor
+export interface LayoutParameters {
+  fit: Fit,
+  alignment: Alignment,
+  minX: number,
+  minY: number,
+  maxX: number,
+  maxY: number
+}
+
+// Alignment options for Rive animations in a HTML canvas
+export class Layout {
+  public fit: Fit;
+  public alignment: Alignment;
+  public minX: number;
+  public minY: number;
+  public maxX: number;
+  public maxY: number;
+
+  constructor(
+    fit: Fit = Fit.None,
+    alignment: Alignment = Alignment.Center,
+    minX: number = 0,
+    minY: number = 0,
+    maxX: number = 0,
+    maxY: number = 0
+  ) {
+    this.fit = fit;
+    this.alignment = alignment;
+    this.minX = minX;
+    this.minY = minY;
+    this.maxX = maxX;
+    this.maxY = maxY;
+  }
+
+  // Alternative constructor to build a Layout from an interface/object
+  static new({fit, alignment, minX, minY, maxX, maxY}: LayoutParameters) : Layout {
+    return new Layout(fit, alignment, minX, minY, maxX, maxY);
+  }
+
+  // Returns fit for the Wasm runtime format
+  public runtimeFit(rive: any): any {
```

Comment:
I would recommend to string type as much as possible the public methods. Of course this is a long term process, but this will reduce enormously the amount of required documentation.

Reply:
This one is for internal use only, and the type's coming from the Wasm Rive object; struggled with properly typing these.